src/mongocxx/test/uri.cpp

```diff
@@ -295,6 +314,230 @@ TEST_CASE("uri::wait_queue_timeout_ms()", "[uri]") {
                        [](mongocxx::uri& uri) { return uri.wait_queue_timeout_ms(); });
 }
 
+static void compare_elements(bsoncxx::document::element expected_option,
```

Comment:
I would name `require_elements_equal` or `assert_elements_equal` to make it clear that this function can terminate the test.